FuelLabs/fuel-indexer
# Allow for live index to be stopped

Closes #348.

## Changelog
- Adds `IndexStopRequest` struct for stopping a live index
- Adds new `ServiceRequest` enum, and adds `AssetReloadRequest` and `IndexStopRequest` to it
- Adds functionality to API to stop execution of an index
- Removes the `run_once` flag for the service; this will be added back in soon as part of #383 

## Testing Steps

1. Rebuild WASM, if necessary
2. Start web components `bash scripts/utils/refresh_test_db.bash` and `bash/scripts/utils/start_web_components.bash`
3. Start indexer `SQLX_OFFLINE=1 cargo run --bin fuel-indexer -- --manifest packages/fuel-indexer-tests/assets/fuel_indexer_test.yaml`
4. Trigger event and see it's indexed `curl -v -X POST http://127.0.0.1:8000/ping`
5. Delete index request `curl -v -X DELETE http://127.0.0.1:29987/api/index/fuel_indexer_test/index1 -H "Authorization: fuel"` -- you should see something in the service logs stating that the index has been stopped
6. Make another request to index event `curl -v -X POST http://127.0.0.1:8000/ping` -- there should be no activity for that index in the logs and there should be no data added to the database as a result of this call
